sdk/simulator-service/src/strategies/refinanceAnyPair/RefinanceLendingToLendingAnyPair.ts

```diff
@@ -222,62 +183,50 @@ export async function refinanceLendingToLendingAnyPair(
- *    accounting the swap and assuming the worst case scenario on slippage IE max slippage.
- *
- *    We also need to factor in Summer fees ahead of time
+ * EstimateTokenAmountAfterSwap
+ * @description Estimates how much you will recive after swap.
+ *    If target token is the same as source token, we return the same amount.
+ *    When we perform a swap, we need to account for the summer fee,
+ *    and we assume maximum slippage.
  */
-function calculateBorrowAmount(params: {
-  isDebtSwapSkipped: boolean
-  prevDebtAmount: TokenAmount
-  debtSpotPrice: Price
+async function estimateSwapFromAmount(params: {
+  receiveAtLeast: TokenAmount
+  fromToken: Token
   slippage: Percentage
-  summerFee: Percentage
-}): TokenAmount {
-  const { isDebtSwapSkipped, prevDebtAmount, debtSpotPrice, slippage, summerFee } = params
+  swapManager: ISwapManager
+}): Promise<TokenAmount> {
+  const { receiveAtLeast, slippage } = params
 
-  /**
-   * If no swap is required we simply borrow the same amount of debt, and the same asset,
-   * on the target protocol
-   */
-  if (isDebtSwapSkipped) {
-    return prevDebtAmount
+  if (isSameTokens(receiveAtLeast.token, params.fromToken)) {
+    return receiveAtLeast
   }
 
-  /**
-   * Worked Example
-   * @description 3 ETH/ 5000 DAI -> X WTBC / Y USDC
-   *    5000 DAI * (0.98 USDC/DAI) = 4900 USDC
-   *    but this assumes zero price impact
-   *
-   *    5000 DAI * (0.98 USDC/DAI) / (1 - 0.01) = 4949.49 USDC (slippage adjusted borrow amount)
-   *    where 0.01 is 1% slippage
-   *
-   *    (5000 DAI * (0.98 USDC/DAI) / (1 - 0.01)) / (1 - 0.002) = 4959.41 USDC (slippage + summer fee adjusted borrow amount)
-   *    where 0.002 is 20 basis pt fee as an example
-   *
-   *    More generally we'd write this as
-   *    (sourcePositionDebt * targetDebtQuotedInSourceDebtPrice / (one - slippage)) / (one - summer fee) = borrowAmount
-   */
-  const borrowAmount = prevDebtAmount.multiply(debtSpotPrice.value)
-  const borrowAmountAdjustedForSlippage = subtractPercentage(
-    borrowAmount,
-    Percentage.createFrom({
-      value: slippage.value,
-    }),
-  )
-  const borrowAmountAdjustedForSlippageAndSummerFee = subtractPercentage(
-    borrowAmountAdjustedForSlippage,
-    Percentage.createFrom({
-      value: summerFee.value,
-    }),
-  )
+  const spotPrice = (
+    await params.swapManager.getSpotPrice({
+      chainInfo: receiveAtLeast.token.chainInfo,
+      baseToken: params.fromToken,
+      quoteToken: receiveAtLeast.token,
+    })
+  ).price
+
+  const summerFee = await params.swapManager.getSummerFee({
+    from: { token: receiveAtLeast.token },
+    to: { token: params.fromToken },
+  })
+
+  const ONE = new BigNumber(1)
+  /*
+  TargetAmt = SourceAmt * (1 - SummerFee) / (SpotPrice * (1 + Slippage))
+  SourceAmt = TargetAmt * SpotPrice * (1 + Slippage) / (1 - SummerFee) 
+  */
+
+  const sourceAmount = params.receiveAtLeast
```

Comment:
I know we have something similar in v1, but we should strive for not using BigNumber in the simulations if possible. What about:
```
const spotPricePlusSlippage = spotPrice.add(spotPrice.multiply(slippage))

const sourceAmountWithSlippage = params.receiveAtLeast.multiply(pricePlusSlippage)

const percent100 = Percentage.createFrom({ value: 100.0 })
const oneMinusSummerFee = percent100.subtract(summerFee)

const sourceAmount = sourceAmountWithoutSlippage.divide(toAddSummerFee)
```